cpp/include/DataStorm/Node.h

```diff
@@ -140,7 +140,7 @@ namespace DataStorm
          * provided the Node will use the default callback executor that executes callback in a dedicated thread.
          */
         Node(
-            Ice::CommunicatorPtr communicator,
+            const Ice::CommunicatorPtr& communicator,
```

Comment:
Because we always use it as a reference.

Reply:
The signature is not / should not be about the implementation but what we signal to the caller.

Here, const& tells the caller: I am reading this communicator in this ctor but I don't keep a refcount on the communicator. Whereas, by value means "I keep a refcount".

I would expect the Node to keep a refcount on the communicator, possibly indirectly by holding an Ice::InstancePtr.